src/librbd/ImageCtx.cc

```diff
@@ -883,6 +883,16 @@ librados::IoCtx duplicate_io_ctx(librados::IoCtx& io_ctx) {
     return new Journal<ImageCtx>(*this);
   }
 
+  crypto::CryptoInterface* ImageCtx::get_crypto() {
+    std::shared_lock image_locker{image_lock};
+    return crypto;
```

Comment:
These new methods open up the possibility of including the `CryptoInterface` header in the `ImageCtx.cc` file so that you can add `crypt->get()` / `crypto-put()` calls

Reply:
I thought I followed this when I read it the first time.
But now I'm not sure where to call get() and where to call put()

Reply:
@trociny I've added a `get()` call after reading the imagectx crypto.
This ensure that the returned crypto object will not be destroyed.

Reply:
Ah. Indeed. Sorry.